frontend/public/components/environment.jsx

```diff
@@ -285,7 +299,33 @@ export const EnvironmentPage = connect(stateToProps)(
           };
         }),
       ])
-        .then(_.spread((configMaps, secrets) => this.setState({configMaps, secrets})));
+        .then((([configMaps, secrets]) => this.setState({configMaps, secrets})));
+    }
+
+    componentDidUpdate(prevProps) {
+      const { obj, model, impersonate, readOnly } = this.props;
+      if (prevProps.obj !== obj ||
```

Comment:
I don't think this will do a deep comparison. Might need to use `!_.isEqual(prevProps.obj, obj)`